src/protagonist/API/Features/Image/ImageController.cs

```diff
@@ -93,12 +93,13 @@ public async Task<IActionResult> GetImage(int customerId, int spaceId, string im
             hydraAsset = LegacyModeConverter.VerifyAndConvertToModernFormat(hydraAsset);
         }
         
+        hydraAsset.ModelId = imageId;

```

Comment:
Should this happen prior to `VerifyAndConvertToModernFormat` as that method could change `hydraAsset.ModelId`?

Reply:
hrm, I think the issue is that the model id holds the value needed to do validation checks, but it's not filled out until `PutOrPatchAsset`, which happens after validation.  However, looking at `VerifyAndConvertToModernFormat ` it sets the value based on a couple of fields.  Given this would it be better to have it after `VerifyAndConvertToModernFormat `, but also use a null check?